### PR TITLE
[veoh] Accept "videos" URLs; prefer HQ format; get additional metadata

### PR DESCRIPTION
**Pull request by foghawk**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This is an update of [youtube-dl PR 22661](https://github.com/ytdl-org/youtube-dl/pull/22661). It makes the following changes to the veoh extractor:

- Accept `videos` URLs.
- Prefer high quality (current master incorrectly marks "Regular" rather than "HQ" as "best").
- Extract metadata for `age_limit`, `categories`, and `tags`.

It also updates the test cases and removes the defunct `_extract_video` method.

Thank you!

### CODE REVIEW

File: yt_dlp/extractor/veoh.py [hunk 1]
```diff
@@ -5,21 +5,30 @@
...
 
     _TESTS = [{
         'url': 'http://www.veoh.com/watch/v56314296nk7Zdmz3',
-        'md5': '9e7ecc0fd8bbee7a69fe38953aeebd30',
+        'md5': '620e68e6a3cff80086df3348426c9ca3',
         'info_dict': {
             'id': 'v56314296nk7Zdmz3',
             'ext': 'mp4',
             'title': 'Straight Backs Are Stronger',
+            'description': 'At LUMOback, we believe straight backs are stronger. The LUMOback Posture & Movement Sensor: It gently vibrates when you slouch, inspiring improved posture and mobility. Use the app to track your data and improve your posture over time.',
```
Comment (pukkandan) [inline yt_dlp/extractor/veoh.py:22]:
```py
            'description': 'md5:...',
```

File: yt_dlp/extractor/veoh.py [hunk 2]
```diff
@@ -51,30 +60,36 @@ class VeohIE(InfoExtractor):
...
+            'age_limit': 18,
+            'categories': ['technology_and_gaming', 'gaming'],
+            'tags': ['puzzle', 'of', 'flesh'],
         }
+    }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
-        video = self._download_json(
+        json = self._download_json(
```
Comment (pukkandan) [inline yt_dlp/extractor/veoh.py:85]: confusing variable name

Comment (foghawk) [reply yt_dlp/extractor/veoh.py]: Is the problem that it's been parsed and therefore isn't JSON _anymore_, that it doesn't specify what was _in_ the JSON, that it implies that `video` isn't also parsed JSON although it is, or something else?

Is this acceptable? (Would appreciate your reasoning regardless; I looked at usage in other extractors and was not able to determine a consistent style.)

Comment (pukkandan) [reply yt_dlp/extractor/veoh.py]: It's because `json` is also the name of a standard library

Comment (foghawk) [reply yt_dlp/extractor/veoh.py]: Ahhh. Thanks.

**Pull request merged**